src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -49,14 +49,15 @@ const logStateTransition = (
   tookMs: number
 ) => {
   if (newState.logs.length > oldState.logs.length) {
+    const doLog = (level: MigrationLogLevel, message: string) => {
+      if (level === 'error') {
+        logger.error(message);
+      } else {
+        logger[level](message);
```

Comment:
There are only 2 levels supported. why not perform an exhaustive check here?
Should we fix in on the logger level by binding context for all the public methods?

Reply:
> why not perform an exhaustive check here?

Yea, probably better

> Should we fix in on the logger level by binding context for all the public methods

I don't think so. We're using javascript and classes, this is a common problem developers needs to be aware of. Fixing it for this specific class will not prevent to have it appear elsewhere. Also binding comes with a cost.